src/libs/elektra/keyhelpers.c

```diff
@@ -190,12 +190,17 @@ int keyNameIsSystem (const char * name)
 /**
  * @internal
  *
+ * @pre key->meta must be NULL or a valid keyset
+ *
  * clears key (all data members are set to zero)
+ * Initializes an empty metadata keyset if null or clears it.
  */
 int keyInit (Key * key)
 {
-	memset (key, 0, sizeof (struct _Key));
+	if (key->meta) ksClear (key->meta);
+	memset (key, 0, sizeof (struct _Key) - sizeof (KeySet *));
```

Comment:
I am not sure if this is portable (layout of struct might be changed) and it is definitely not friendly for maintenance. Maybe simply remember the value of key->meta with a local variable.

Reply:
Pretty sure, this is as far from portable as it gets. AFAIK C doesn't make any guarantees related to struct layout (unless special compiler attributes are used), so the layout might even be different with different compiler settings on the same machine...